web/src/lib/endpoints.ts

```diff
@@ -5,19 +5,50 @@ export const API = {
 	ME: `/api/me`,
 	ME_NAME: `/api/me/name`,
 
-	ELECTION: `/api/election`,
+	POLL: `/api/poll`,
+	POLL_CURRENT: `/api/poll/current`,
+	POLL_RESULTS: `/api/poll/results`,
+	POLL_SSE: `/api/poll/sse`,
 	ELECTION_STAND: `/api/election/stand`,
-	ELECTION_CURRENT: `/api/election/current`,
-	ELECTION_CURRENT_VOTE: `/api/election/current/vote`,
-	ELECTION_SSE: `/api/election/sse`,
-	ELECTION_RESULTS: `/api/election/results`,
+	ELECTION_VOTE: `/api/election/vote`,
+	REFERENDUM_VOTE: `/api/referendum/vote`,
 
+	ADMIN_POLL: `/api/admin/poll`,
+	ADMIN_POLL_PUBLISH: `/api/admin/poll/publish`,
+	ADMIN_POLL_SSE: `/api/admin/poll/sse`,
 	ADMIN_ELECTION: `/api/admin/election`,
-	ADMIN_ELECTION_SSE: `/api/admin/election/sse`,
 	ADMIN_ELECTION_START: `/api/admin/election/start`,
 	ADMIN_ELECTION_STOP: `/api/admin/election/stop`,
-	ADMIN_ELECTION_PUBLISH: `/api/admin/election/publish`,
+	ADMIN_REFERENDUM: `/api/admin/referendum`,
+	ADMIN_REFERENDUM_START: `/api/admin/referendum/start`,
+	ADMIN_REFERENDUM_STOP: `/api/admin/referendum/stop`,
 	ADMIN_USER: `/api/admin/user`,
 	ADMIN_USER_DELETE: `/api/admin/user/delete`,
 	ADMIN_USER_RESTRICT: `/api/admin/user/restrict`,
 } as const;
+
+export const PollTypeId = {
+	ELECTION: 1,
+	REFERENDUM: 2,
+}
+
+type EndpointType = "vote" | "create" | "start" | "stop";
+
+export const getEndpointForPollType = (endpointType: EndpointType, pollType: number): string | undefined => {
+	const endpoints = {

```

Comment:
Consider dynamically constructing the object access, e.g. 
```
let endpoint = "";
bool needsAdmin = ['create',  'start', 'stop'].includes(endPointType);
if (needsAdmin) {
  endpoint += "ADMIN_";
}
endpoint += pollType; // pollType: 'REFERENDUM' | 'ELECTION'
endpoint += endpointType.toUpperCase(); // or refactor the type to uppercase
return API[endpoint];
```

Reply:
IMO this makes it harder to read for not much benefit. Although I guess it might be useful should there ever be any more poll types or operations on polls